### PR TITLE
blender: enable path guiding support; openpgl: init at 0.5.0

### PR DESCRIPTION
**Pull request by amarshall**
## Description of changes

See commit messages.

## Things done

Built Blender and verified path guiding option is available (only supported with CPU rendering) and rendering default cube with it enabled succeeds.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24560,6 +24560,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) PCSC;
   };
 
+  openpgl = callPackage ../development/libraries/openpgl { };
```
Comment (nagy) [inline pkgs/top-level/all-packages.nix:24563]: Can you use the new by-name structure https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#name-based-package-directories ?

**Review by nagy — COMMENTED** [review]
How much does this increase the closure size?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (nagy) [reply pkgs/top-level/all-packages.nix:24563]: The point of the by-name structure is, that you don't need this line anymore. but that does not play well with the other suggestion about the tbb override

Comment (veprbl) [reply pkgs/top-level/all-packages.nix:24563]: Need to fix CI checks now.

File: pkgs/by-name/op/openpgl/dev.patch [hunk 2]
```diff
@@ -0,0 +1,38 @@
+diff --git a/cmake/FindTBB.cmake b/cmake/FindTBB.cmake
```
Comment (veprbl) [inline pkgs/by-name/op/openpgl/dev.patch:1]: Unused file. Please, remove!

**Review by nagy — APPROVED** [review]

File: pkgs/by-name/op/openpgl/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, cmake
+, fetchFromGitHub
+, ninja
+, stdenv
+, tbb
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```
[the remark this answers is not in the document]
Comment (veprbl) [reply pkgs/by-name/op/openpgl/package.nix:9]: No. Do you have numbers to prove fixed points are free? There is no advantage for using it here.

[the remark this answers is not in the document]
Comment (veprbl) [reply pkgs/by-name/op/openpgl/package.nix:9]: Okay. Will be looking forward to a treewide switch.

**Pull request merged**